hail/src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -653,6 +655,30 @@ object Interpret {
       case Die(message, typ) => fatal(message)
       case ir@ApplyIR(function, functionArgs, conversion) =>
         interpret(ir.explicitNode, env, args, agg)
+      case ApplySpecial("||", Seq(left_, right_)) =>
```

Comment:
cc @catoverdrive - I had to add these rules to make python tests pass after fixing this bug.

Reply:
ah, good call. That should let us fix the places where we want to check this in other tests, too.